pymatgen/transformations/tests/test_advanced_transformations.py

```diff
@@ -614,6 +615,30 @@ def test_apply_transformation(self):
         self.assertDictEqual(output[-1].species.as_dict(),
                              {'Ni': 0.5, 'Ba': 0.5})
 
+class CubicSupercellTransformationTest(PymatgenTest):
+
+    def test_gen_scaling_matrix(self):
```

Comment:
Look at the tests for other transformers. You should be testing the `apply_transformation` function. You are also not validating that the output of the transformation is correct.

See the `assert` functions in other tests.

Reply:
So I wasn't sure about how to test 'apply_transformation' because I don't necessarily know what the 'correct' output is. For example, if I changed the line 'target_sc_size += 0.1' to '+= 0.01', I might find a better transformation that the step size of '0.1' was too big to catch. Should I simply test that apply_transformation produces a valid Pymatgen structure? Or maybe that the supercell that is at least as cubic as what my current version yields? Even then I will need to somewhat arbitrarily define some measure of how cubic something is.

Reply:
The cubic transformation is deterministic, so you can just check the lattice parameters match what you expect and that the cell has the correct number of atoms.

For the perturb sites, you should check:
- That the number of structures you asked for is returned
- That the cell has been perturbed, e.g. test that some of the sites are not in their original positions.